Added gradle home mount point

ntrp opened the pull request.

ntrp committed:
Added gradle home mountpoint

stromvirvel commented on init-docker-buildenv.sh:
```diff
@@ -113,12 +113,22 @@ echo "init-docker-buildenv: starting job in a new docker container"
+     "${docker_image}" "/build/docker/start.sh"
 else
   user_host=`whoami`
   mkdir -p ~/.m2
```
Can you please also create the .gradle folder? Because if folder does not exist, the docker run command will fail.

stromvirvel replied:
Just thinking about moving the "mkdir" commands above the IF-statement, so that the folders are created, no matter if it runs on a bambooagent or not ...
Just thinking about a bamboo agents which has never used before. There won't be a .gradle folder and you will never be able to run your gradle tasks inside a docker container.

ntrp replied:
to what does `~bambooagent/.m2/` point? How comes it's not `~/.m2/`

stromvirvel replied:
`~foo` points to foo's home folder.
`~` points to the current user's home folder.

bamboo will run the script as `bambooagent` so there should be no difference using `~bambooagent` or `~`. I'm not sure why I've chosen the first one. Probably to make thinks more clear. But I think it's okay to leave it as it is for now.

stromvirvel replied:
Otherwise, if you want to change it I'd kindly ask you to test it before locally and on a bamboo agent :) Thank you.

ntrp committed:
adding directory creation

For now let's go with the safe version, I don't have time to test changing the directories etc. at the moment so I prefer to merge at least this. I will experiment a little maybe in the near future.
